src/security-rules/security-rules.ts

```diff
@@ -143,9 +115,59 @@ export class SecurityRules implements FirebaseServiceInterface {
     return this.getRulesetForRelease(SecurityRules.CLOUD_FIRESTORE);
   }
 
+  /**
+   * Creates a `RulesFile` with the given name and source. Throws if any of the arguments are invalid. This is a
+   * local operation, and does not involve any network API calls.
+   *
+   * @param {string} name Name to assign to the rules file.
+   * @param {string|Buffer} source Contents of the rules file.
+   * @returns {RulesFile} A new rules file instance.
+   */
+  public createRulesFileFromSource(name: string, source: string | Buffer): RulesFile {
+    if (!validator.isNonEmptyString(name)) {
+      throw new FirebaseSecurityRulesError(
+        'invalid-argument', 'Name must be a non-empty string.');
+    }
+
+    let content: string;
+    if (validator.isNonEmptyString(source)) {
+      content = source;
+    } else if (validator.isBuffer(source)) {
+      content = source.toString('utf-8');
+    } else {
+      throw new FirebaseSecurityRulesError(
+        'invalid-argument', 'Source must be a non-empty string or a Buffer.');
+    }
+
+    return {
+      name,
+      content,
+    };
+  }
+
+  /**
+   * Creates a new `Ruleset` with one or more rules files.
+   *
+   * @param {RulesFile} file Rules file to include in the new Ruleset.
+   * @param {RulesFile[]} additionalFiles Additional rules files to be included in the new Ruleset.
+   * @returns {Promise<Ruleset>} A promise that fulfills with the newly created Ruleset.
+   */
+  public createRuleset(file: RulesFile, ...additionalFiles: RulesFile[]): Promise<Ruleset> {
```

Comment:
I'm not sure we should allow developers to upload multiple rules files. It is not supported at the moment, and we'll probably want to re-think this API when it comes to adding support for multiple files.

Reply:
That's a good point. Removed `additionalFiles` from the signature and changed the implementation as needed. We can add it back in the future as a non-breaking change if necessary.